pandas/util/_decorators.py

```diff
@@ -260,40 +260,37 @@ def doc(*args: Union[str, Callable], **kwargs: str) -> Callable[[F], F]:
     ----------
     *args : str or callable
         The string / docstring / docstring template to be appended in order
-        after default docstring under function.
+        after default docstring under callable.
     **kwags : str
         The string which would be used to format docstring template.
     """
 
-    def decorator(func: F) -> F:
-        @wraps(func)
-        def wrapper(*args, **kwargs) -> Callable:
-            return func(*args, **kwargs)
-
+    def decorator(call: Callable) -> Callable:
```

Comment:
`call` doesn't seem very descriptive to me, may be `decorated_obj`, `func_or_class` or something like that?

Is a class of type `Callable`, even if it doesn't implement `__call__`? @simonjayhawkins may be you can help, the parameter can be any function or any class, is `Callable` the appropriate type?

Reply:
IIUC then this seems OK since we are passing a class type not instance, and a class type is callable in order to instantiate.

```
from typing import Callable

class Foo:
    pass

def func(call:Callable):
    pass

func(Foo)

reveal_type(Foo)
```
gives
type-test.py:11: note: Revealed type is 'def () -> type-test.Foo'

and no errors. The class type seems to be treated as a function that returns an instance by mypy.